catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/WhenBrickTest.java

```diff
@@ -0,0 +1,115 @@
+
+import android.support.annotation.StringRes;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.BroadcastReceiverBrick;
+import org.catrobat.catroid.content.bricks.PlaceAtBrick;
+import org.catrobat.catroid.content.bricks.WhenBackgroundChangesBrick;
+import org.catrobat.catroid.content.bricks.WhenClonedBrick;
+import org.catrobat.catroid.content.bricks.WhenConditionBrick;
+import org.catrobat.catroid.content.bricks.WhenStartedBrick;
+import org.catrobat.catroid.content.bricks.WhenTouchDownBrick;
+import org.catrobat.catroid.physics.content.bricks.CollisionReceiverBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickCoordinatesProvider;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.junit.runners.Parameterized.Parameters;
+
+@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
+public class WhenBrickTest {
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Parameters(name = "{2}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{R.string.brick_when_started, WhenStartedBrick.class, "WhenStartedBrick"},
+				{R.string.brick_when_becomes_true, WhenConditionBrick.class, "WhenConditionBrick"},
+				{R.string.brick_when_cloned, WhenClonedBrick.class, "WhenClonedBrick"},
+				{R.string.brick_when_background, WhenBackgroundChangesBrick.class, "WhenBackgroundChangesBrick"},
+				{R.string.brick_when_touched, WhenTouchDownBrick.class, "WhenTouchDownBrick"},
+				{R.string.brick_broadcast_receive, BroadcastReceiverBrick.class, "BroadcastReceiverBrick"},
+				{R.string.brick_collision_receive, CollisionReceiverBrick.class, "CollisionReceiverBrick"}
+		});
+	}
+
+	@Parameter
+	public @StringRes int whenBrickNameId;
+
+	@Parameter(1)
+	public Class<?> whenBrickClass;
+
+	@Parameter(2)
+	public String testName;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("WhenBrickTest");
+
+		if (whenBrickClass.getName().contains("CollisionReceiverBrick")) {
```

Comment:
why not go with
whenBrickClass.equals(WhenBrick.class)
and skip the string comparison